### PR TITLE
Tweak onboarding errors

### LINKED ISSUE #328: Need more helpful error messages during onboarding
- Device: iPhone 8 plus
- iOS version: Latest
- Delta Chat version: Latest
- Expected behavior: 
  - Log in, or get an error message that tells me that I need to change some email settings in order for it to work.
- Actual behavior: Receive bizarre error messages. No clear fixes that I can provide. User can only just close the app and give up on it.
- Steps to reproduce the problem:  

Do not change any app advanced settings or default email settings for these providers. Just simply enter your emails and passwords.

- Screenshots:

Here is the error I got when I tried my iCloud email:
icloud

Here is the error I got when I tried my Gmail email:
gmail

### PR DESCRIPTION
**Pull request by r10s**
closes #328

### CODE REVIEW

File: deltachat-ios/DC/events.swift [hunk 1]
```diff
@@ -13,6 +13,13 @@ let dcNotificationContactChanged = Notification.Name(rawValue: "MrEventContactsC
 @_silgen_name("callbackSwift")
 
 public func callbackSwift(event: CInt, data1: CUnsignedLong, data2: CUnsignedLong, data1String: UnsafePointer<Int8>, data2String: UnsafePointer<Int8>) -> UnsafePointer<Int8>? {
+    if event >= DC_EVENT_ERROR && event <= 499 {
```
Comment (cyBerta) [inline deltachat-ios/DC/events.swift]: I guess the idea here is to suppress the error events that are in the range of 400 - 499 to be shown to the user, right? 
This condition includes not only DC_EVENT_ERROR_NETWORK, but also also DC_EVENT_ERROR_SELF_NOT_IN_GROUP. Is that ok?

Comment (r10s) [reply deltachat-ios/DC/events.swift]: yip, that is okay for now. it has to be settled in deltachat.h, however, i do not want to wait for that here.

Comment (r10s) [reply deltachat-ios/DC/events.swift]: and: the errors are not suppressed.
they are logged as before logger.error("event: \(s)") - and moreover, the AppDelegate.lastErrorString is set for _all_ errors (was not true for DC_EVENT_ERROR_NETWORK before), this was one of the reasons for #328

**Comment by trosel** [comment]
@r10s This is an improvement, but the end user still has no actionable takeaway from this error.
```
DeltaChat: "Could not log in"
User:      "Okay, what did I do wrong?"
DeltaChat: ...
User:      *deletes app*
```

**Comment by r10s** [comment]
also fixes https://github.com/deltachat/deltachat-ios/issues/281

**Pull request merged**